### PR TITLE
[VarDumper][HtmlDumper] Take care of maxDepth display option if provided

### PR DESCRIPTION
**Pull request by ngandemer**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tests pass? | yes
| Tickets       | ~ 
| License       | MIT
| Doc PR        | ~ 

When using HtmlDumper for a deep object and with option maxDepth=n, the dumped object is printed fully, and doesn't care of maxDepth option. A a consequence, browser can be blocked while printing the dumped object (seen in profiler).

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
The `maxDepth` option in `HtmlDumper` is meant to configure the level at which nested structures are *toggled*.

Limiting the size of the data should be done *before* the `Data` object hits the dumper, using `Data::withMaxDepth` as you spotted.
Another way to limit the size is to use `setMaxItems` on the cloner.
There is a config option in Symfony to define this:
```yaml
debug:
    max_items: 2500
```

TL;DR, this is not the right patch. I don't know your use case so I can't suggest more alternatives.

**Comment by nicolas-grekas** [comment]
Closing as explained.
